spec/datadog/core/environment/process_spec.rb

```diff
@@ -147,4 +148,107 @@ def reset_serialized!
       end
     end
   end
+  describe '::tags_array' do
+    subject(:tags_array) { described_class.tags_array }
+
+    shared_context 'with mocked process environment' do
+      let(:pwd) { '/app' }
+
+      around do |example|
+        @original_0 = $0
+        $0 = program_name
+        example.run
+        $0 = @original_0
+      end
+
+      before do
+        allow(Dir).to receive(:pwd).and_return(pwd)
+        allow(File).to receive(:expand_path).and_call_original
+        allow(File).to receive(:expand_path).with('.').and_return('/app')
+        reset_memoized_variables!
+      end
+
+      after do
+        reset_memoized_variables!
+      end
+    end
```

Comment:
This shared context is the same as https://github.com/DataDog/dd-trace-rb/blob/18e2b12ebbf4dbcbc590e3739aedfd2c91ae26de/spec/datadog/core/environment/process_spec.rb#L14

and equivalent to: https://github.com/DataDog/dd-trace-rb/blob/18e2b12ebbf4dbcbc590e3739aedfd2c91ae26de/spec/datadog/core/remote/client_spec.rb#L45-L66
I suggest we share this context :)
The shared context is kind of complicated, so I think duplicating it will cause trouble in the future, in case of drift.

I recommend moving it to the top-level of this file, declaring it as `RSpec.shared_context 'with mocked process environment'`.

I can then be used in this file, and imported in `spec/datadog/core/remote/client_spec.rb`.